packages/sds/src/message_channel/sorted_array_base.ts

```diff
@@ -0,0 +1,306 @@
+type SortedArrayInterface<T> = Omit<Array<T>, "reverse" | "sort">;
+
+/**
+ * Base class that implements a sorted array interface.
+ * Automatically maintains sort order after any mutating operation.
+ */
+export abstract class SortedArrayBase<T> implements SortedArrayInterface<T> {
```

Comment:
i would reconsider adding this code to our repository (and then having to maintain/test it) versus introducing a small fixed dependency (maybe a sub-package of lodash? https://lodash.com/ )